inductiva/simulators/simulator.py

```diff
@@ -6,12 +6,30 @@
 from inductiva.utils import files
 
 
+def mpi_enabled(cls):
+    """Class decorator that adds MPICluster to the supported resources.
+    """
+    supports = getattr(cls, "_supported_resources", None)
+    cls._supported_resources = supports | {resources.MPICluster}  # pylint: disable=protected-access
+
+    return cls
+
+
 class Simulator(ABC):
     """Base simulator class."""
 
+    _supported_resources = {
+        resources.MachineGroup, resources.ElasticMachineGroup
+    }
+
     def __init__(self):
         self.api_method_name = ""
 
+    @classmethod
+    def get_supported_resources(cls):
+        """Get the supported computational resources for this simulator."""
+        return tuple(cls._supported_resources.copy())
```

Comment:
why return a tuple? We're already returning a copy of the set which is guaranteed to be unique.

Either
```suggestion
        return tuple(cls._supported_resources)
or
        return cls._supported_resources.copy()
```